src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -4622,6 +4622,18 @@ internal Token NextToken()
                         return ScanHereStringExpandable();
                     }
 
+                    if (ExperimentalFeature.IsEnabled("PSGeneralizedSplatting"))
+                    {
+                        if (c1 == '@')
+                        {
+                            var c2 = PeekChar();
```

Comment:
Rather than doing a lookahead here, the tokenizer shouldn't be contextual. Instead, the token should just be `@@{`. That accurately represents the token the user sees, the lexical work done here and prevents the parser from having to skip a brace token.

It also means that the few tools that rely on PowerShell tokens don't get confused by an open brace token, rather than a single token representing `@@{`

Reply:
Ok, I've implemented your suggested change.